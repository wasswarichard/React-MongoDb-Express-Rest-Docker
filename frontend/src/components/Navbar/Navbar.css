header {
    background: #ffffff;
    display: flex;
    justify-content: space-between;
    padding: 1rem 1.5rem;
    box-shadow: 2px 2px 5px rgba(0,0,0,0.2);
    position: fixed;
    left: 300px;
    width: calc(100% - 300px);
    top: 0;
    z-index: 100;
    transition: left 300ms;
}
header h2 {
    color: #222;
}

header label span {
    font-size: 1.7rem;
    padding-right: 1rem;
}

.user-wrapper {
    display: flex;
    align-items: center;
}

.user-wrapper img {
    border-radius: 50%;
    margin-right: 1rem;
}

.user-wrapper small {
    display: inline-block;
    color: var(--text-grey);
}
.user-logout {
    margin-left: 30px;
}

.logout {
    text-decoration: underline;
    cursor: pointer;
}