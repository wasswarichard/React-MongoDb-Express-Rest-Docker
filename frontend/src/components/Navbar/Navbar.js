import React from "react";
import profilePicture from "../../images/richard.jpg"
import 'bootstrap/dist/css/bootstrap.css';
import "./Navbar.css"
import Grid from "@material-ui/core/Grid";
import config from "../../helpers/config.json";
import {store} from "../../state/store/store";

const Navbar = () => {
    const logout = () => {
        const authentication = store.getState().session
        fetch(`${config.apiUrl}/api/removesession`, {
            headers: {
                authorization: authentication.accessToken,
                "x-refresh": authentication.refreshToken
            },
            method: "POST",
        }).then(response => {
            if (response.status === 200) {
                localStorage.removeItem('session');
                window.location.href = '/';
            }
        })
    }
    return (
        <div className="main-content">
            <header>
                <h2>
                    <label htmlFor="nav-toggle">
                        <span className="las la-bars"></span>
                    </label>
                    Dashboard
                </h2>
                <div className="user-wrapper">
                    <img src={profilePicture} width="40px" height="40px" alt=""/>
                    <div>
                        <h5> Richard W</h5>
                        <small> Super admin</small>
                    </div>
                    <div className="user-logout">
                        <Grid container>
                            <Grid item xs>
                                <a onClick={event => logout(event)} className="logout">
                                    log out
                                </a>
                            </Grid>
                        </Grid>
                    </div>
                </div>
            </header>
        </div>

    )
}
export default Navbar;