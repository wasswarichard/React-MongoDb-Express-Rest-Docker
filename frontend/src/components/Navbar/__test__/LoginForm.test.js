import React from "react";
import ReactDOM from "react-dom";
import Navbar from "../Navbar";
import { cleanup} from "@testing-library/react";
import TestRenderer from 'react-test-renderer';

afterEach(cleanup)
it('should render without crashing', function () {
    const div = document.createElement("div")
    ReactDOM.render(<Navbar></Navbar>, div)
});

it('should match snapshot', function () {
    const tree = TestRenderer.create(<Navbar></Navbar>).toJSON();
    expect(tree).toMatchSnapshot();

});
