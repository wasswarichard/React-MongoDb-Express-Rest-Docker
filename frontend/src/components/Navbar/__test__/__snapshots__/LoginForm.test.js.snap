// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="main-content"
>
  <header>
    <h2>
      <label
        htmlFor="nav-toggle"
      >
        <span
          className="las la-bars"
        />
      </label>
      Dashboard
    </h2>
    <div
      className="user-wrapper"
    >
      <img
        alt=""
        height="40px"
        src="richard.jpg"
        width="40px"
      />
      <div>
        <small>
           Super admin
        </small>
      </div>
      <div
        className="user-logout"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <a
              className="logout"
              onClick={[Function]}
            >
              log out
            </a>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
